### PR TITLE
Add switch for Shelly script

### PR DESCRIPTION
**Pull request by chemelli74**
## Proposed change

Add info about switch for Shelly script

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/108171
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **Documentation**
	- Added a new section on "Scripts (generation 2 and 3)" for the Shelly integration, detailing switch entities for controlling device scripts.
	- Clarified firmware compatibility requirements for generation 1 and generation 2 devices.
	- Expanded the known issues section to include specifics on unsupported devices and operational requirements for battery-powered devices, including wake-up needs before setup.

### CODE REVIEW

File: source/_integrations/shelly.markdown [hunk 1]
```diff
@@ -330,6 +330,10 @@ The integration supports the following virtual components:
 - `text` in `field` mode, for which a `text` platform entity is created
 - `text` in `label` mode, for which a `sensor` platform entity is created
 
+## Scripts
```
Comment (thecode) [inline source/_integrations/shelly.markdown]:
```suggestion
## Scripts (generation 2 and 3)
```

**Review by thecode — DISMISSED** [review]
In other places we specify to which Gen it is relevant to

File: source/_integrations/shelly.markdown [hunk 2]
```diff
@@ -330,6 +330,10 @@ The integration supports the following virtual components:
...
+
+For each device script, the integration creates a switch entity that allows you to control the script. These entities are disabled by default.
```
Comment (bieniu) [inline source/_integrations/shelly.markdown]:
```suggestion
For each device script, the integration creates a `switch` entity that allows you to control the script. These entities are disabled by default.
```

File: source/_integrations/shelly.markdown (hunk 1 above)
Comment (chemelli74) [reply source/_integrations/shelly.markdown]: I just mutuate what we did for Virtual Components that are lacking the gen

**Review by bieniu — COMMENTED** [review]
Small suggestion

**Review by bieniu — APPROVED** [review]
LGTM :+1:

**Review by thecode — APPROVED** [review]

**Pull request merged**